### PR TITLE
feat: llm intent classification (PL-844)

### PR DESCRIPTION
**Pull request by theProf**
**Fixes or implements PL-844**

### Brief description. What is this change?

Refactor of llm/hybrid intent classification

### Implementation details. How do you make this change?

Refactored prediction to class and separate out intent prediction and slot filling

### Related PRs

- https://github.com/voiceflow/general-runtime/pull/716

### Checklist

- [ ] manual testing
- [ ] Appropriate tests have been written

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -186,7 +187,8 @@
     "test:integration": "NODE_ENV=test nyc --report-dir=nyc_coverage_integration ts-mocha --paths --opts ./config/tests/mocha.opts 'tests/**/*.it.ts'",
     "test:run": "NODE_ENV=test nyc ts-mocha --paths --opts ./config/tests/mocha.opts 'tests/**/*.{unit,it}.ts'",
     "test:single": "NODE_ENV=test ts-mocha --paths --opts ./config/tests/mocha.opts",
-    "test:unit": "NODE_ENV=test nyc --report-dir=nyc_coverage_unit ts-mocha --paths --opts ./config/tests/mocha.opts 'tests/**/*.unit.ts'"
+    "test:unit": "NODE_ENV=test nyc --report-dir=nyc_coverage_unit ts-mocha --paths --opts ./config/tests/mocha.opts 'tests/**/*.unit.ts'",
+    "test:greg": "NODE_ENV=test ts-mocha --paths --opts ./config/tests/mocha.opts --watch"
```
[the remark this answers is not in the document]
Comment (theProf) [reply package.json:191]: nope, that snuck in, thanks

File: lib/services/classification/predictor.class.ts [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+      .generateCompletion({
+        workspaceID: this._props.workspaceID,
+        prompt,
+        params: {
+          // TODO: models are different between ml gateway sdk and dtos package
+          model: this._settings.params.model as any,
+          temperature: this._settings.params.temperature,
+        },
+        options: {
+          // TODO: remove magic number
```
Comment (trs) [inline lib/services/classification/predictor.class.ts:242]: A few outstanding todos.

File: lib/services/classification/prompt-wrapper-executor.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      await mainModule.evaluate({
+        reference: true,
+        timeout: 1000,
+      });
+    } catch (err) {
+      reject(err);
+    }
+  });
+
+  // if (!isStringReference(result)) {
```
[the remark this answers is not in the document]
Comment (theProf) [reply lib/services/classification/prompt-wrapper-executor.ts:67]: i'll need help with this file. The original code we paired on had it returning a `Reference` and this checked if it referenced a string value before derefing, but the `result` is just the value returned and not a reference. I'm not as confident with the isolated-vm code given the maneuvering that went on to get execution in the first place, if someone would be able to lend a hand here

File: lib/services/nlu/nlc.ts [hunk 4]
```diff
@@ -36,7 +36,10 @@ export const registerIntents = (
   dmRequest?: BaseRequest.IntentRequestPayload
```
[the remark this answers is not in the document]
Comment (theProf) [reply lib/services/nlu/nlc.ts:36]: this is all still used for nlc, we use `handleNLCCommand`

File: tests/lib/services/classification/prompt-wrapper-executor.test.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import { expect } from 'chai';
```
Comment (trs) [inline tests/lib/services/classification/prompt-wrapper-executor.test.ts:1]: This file isn't being tested, it needs to end in `.unit.ts`.

These tests are also failing, the `executePromptWrapper` result is always `undefined`.

**Review by trs — CHANGES_REQUESTED** [review]

File: lib/services/classification/prompt-wrapper-executor.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
+import { randomUUID } from 'crypto';
+import ivm from 'isolated-vm';
+import withResolvers from 'promise.withresolvers';
+
+import { EXECUTION_TIMEOUT_MS, MEMORY_LIMIT_MB } from './promot-wrapper.const';
```
Comment (DecathectZero) [inline lib/services/classification/prompt-wrapper-executor.ts:5]: typo in the file name

**Comment by trs** [comment]
Split up:

https://github.com/voiceflow/general-runtime/pull/743
https://github.com/voiceflow/general-runtime/pull/742
https://github.com/voiceflow/general-runtime/pull/744
https://github.com/voiceflow/general-runtime/pull/745
https://github.com/voiceflow/general-runtime/pull/746
https://github.com/voiceflow/general-runtime/pull/747
https://github.com/voiceflow/general-runtime/pull/748

**Pull request closed without merging**